cirq-core/cirq/sim/simulation_state.py

```diff
@@ -166,6 +166,15 @@ def create_merged_state(self) -> Self:
         """Creates a final merged state."""
         return self
 
+    def add_qubits(self: Self, qubits: Sequence['cirq.Qid']) -> Self:
+        """Add qubits to a new state space and take the kron product."""
+        new_space = type(self)(qubits=qubits)
+        return self.kronecker_product(new_space, inplace=True)
+
+    def remove_qubits(self: Self, qubits: Sequence['cirq.Qid']) -> Self:
+        """Remove qubits from the state space."""
```

Comment:
I think you need a check for empty `qubits` here, and exit early if so. Otherwise `factor` can change the phase. Alternately, do this check within the `factor_state_vector` and `factor_density_matrix` functions in the `transformers` package that get called under the hood.

The latter is probably the better option for consistency's sake, though it's worth testing whether the added branch causes any notable perf changes in the nominal case like factoring a 2-axis state vector into two single-axis ones, since those are core functions.

Reply:
Also add inputs/outputs to the docstring and a note that only subclasses that support `kronecker_product` will support this function.